src/main/java/com/google/googleidentity/oauth2/util/OAuth2Utils.java

```diff
@@ -44,4 +43,51 @@
         return ImmutableSet.copyOf(scopes);
 
     }
+
+    /**
+     * Get UserSession from HttpSession
+     *
+     * @param request
+     * @return
+     */
+    public static UserSession getUserSession(HttpServletRequest request){
+        UserSession userSession =
+                (UserSession) request.getSession().getAttribute("user_session");
```

Comment:
nit: extract "user_session" and "client_session" attribute names to private constants.